### PR TITLE
Initial SFA.DAS.Vacancy.Analytics.Client

### PR DESCRIPTION
**Pull request by sachpatel**

### CODE REVIEW

File: SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/Events/ApprenticeshipApplicationSubmittedEvent.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System;
+
+namespace Esfa.Vacancy.Analytics.Events
+{
+    public sealed class ApprenticeshipApplicationSubmittedEvent : VacancyEvent
+    {
+        public override string EventType => this.GetType().Name;
```
Comment (leewadhams) [inline SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/Events/ApprenticeshipApplicationSubmittedEvent.cs]: Is `nameof(ApprenticeshipApplicationSubmittedEvent)` a better option?

File: SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/Events/ApprenticeshipApplicationSubmittedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        public override DateTime EventTime { get; protected set; }
```
Comment (leewadhams) [inline SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/Events/ApprenticeshipApplicationSubmittedEvent.cs]: Why `protected set` ??

File: SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <EmbedUntrackedSources>false</EmbedUntrackedSources>
+    <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
+    <PackageLicenseUrl>https://github.com/SkillsFundingAgency/das-shared-packages/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/SkillsFundingAgency/das-shared-packages/tree/master/SFA.DAS.Vacancy.Analytics.Client</PackageProjectUrl>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <PackageReleaseNotes>Initial package for publish events from the Find an apprenticeship service.</PackageReleaseNotes>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All" />
```
Comment (leewadhams) [inline SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client.csproj]: What's this used for?

File: SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/VacancyEventClient.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            await PublishEventAsync(evt);
+        }
+
+        private async Task PublishEventAsync(VacancyEvent evt)
+        {
+            var evtData = new EventData(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(evt)));
+
+            try
+            {
+                var client = GetClient();
```
Comment (leewadhams) [inline SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/VacancyEventClient.cs]: Might not be an issue but will there be any issues with port exhaustion or something similar with the client created on every call?

Comment (sachpatel) [reply SFA.DAS.Vacancy.Analytics.Client/SFA.DAS.Vacancy.Analytics.Client/VacancyEventClient.cs]: Don't believe so, I have created and thrown 100 events in like a minute and had no issue. Will ahve to see when we test it out in wild on TEST. Will do more testing.

**Pull request merged**